File: Freedom/微能量/基本类库/MicroEnergyTabBarController.m
//  MicroEnergyTabBarController.m
//  Created by Super on 16/8/19.
//  Copyright © 2016年 Super. All rights reserved.
//
#import "MicroEnergyTabBarController.h"
#import "UIImage+expanded.h"
#import "EnergyNavigationController.h"
#import "EnergySuperMarketTabBarController.h"
@interface MicroEnergyTabBarController ()<UITabBarControllerDelegate>{
    EnergySuperMarketTabBarController *myTabBar;
}
@end
@implementation MicroEnergyTabBarController
- (void)viewDidLoad {
    [super viewDidLoad];
    self.tabBar.barTintColor = RGBACOLOR(59, 59, 59,1);
    //通过设置文本属性来设置字体颜色
    for(int i=0;i<self.childViewControllers.count;i++){
        NSMutableDictionary *attM = [NSMutableDictionary dictionary];
        [attM setObject:[UIColor orangeColor] forKey:NSForegroundColorAttributeName];
        UIViewController *s = self.childViewControllers[i];
        [s.tabBarItem setTitleTextAttributes:attM forState:UIControlStateSelected];
        s.tabBarItem.image = [s.tabBarItem.image imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
        s.tabBarItem.selectedImage = [s.tabBarItem.selectedImage imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
        s.tabBarItem.tag = i;
    }
    myTabBar=[EnergySuperMarketTabBarController sharedRootViewController];
    myTabBar.hidesBottomBarWhenPushed = YES;
    myTabBar.superTabbar = self;
}
-(void)tabBar:(UITabBar *)tabBar didSelectItem:(UITabBarItem *)item{
    if(item.tag!=3){
         myTabBar.backTab = item.tag;
    }else{
        EnergyNavigationController *navi = self.childViewControllers[3];
        [navi pushViewController:myTabBar animated:YES];
    }
}
-(void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
}
@end

File: Freedom/微能量/基本类库/MicroEnergyTabBarController.swift
//
//  MicroEnergyTabBarController.swift
//  Freedom
import UIKit
import XExtension
//class JuheTabBar: UITabBar {
//    var centerButton = UIButton()
//    override init(frame: CGRect) {
//        super.init(frame: frame)
//        centerButton.setImage(UIImage(named: "juhetab3"), for: .normal)
//        centerButton.frame = CGRect(x: 0, y: 0, width:APPW/5, height:50)
//        centerButton.layer.cornerRadius = 25
//        centerButton.clipsToBounds = true
//        addSubview(centerButton)
//    }
//    required init?(coder aDecoder: NSCoder) {
//        fatalError("init(coder:) has not been implemented")
//    }
//    override func layoutSubviews() {
//        super.layoutSubviews()
//        centerButton.center = CGPoint(x: bounds.size.width * 0.5, y: bounds.size.height * 0.5 - 12)
//        let tabbarButtonW: CGFloat = bounds.size.width / 5
//        var buttonIndex: CGFloat = 0
//        for view in subviews {
//            if view.isKind(of: NSClassFromString("UITabBarButton")! ) {
//                view.frame = CGRect(x: buttonIndex * tabbarButtonW,y:view.frame.origin.y, width: tabbarButtonW, height:view.bounds.size.height)
//                buttonIndex += 1
//                if buttonIndex == 3 {
//                    //buttonIndex++;
//                    view.isHidden = true
//                }
//            }
//        }
//    }
//}
//class JuheDataTabBarController: BaseTabBarViewController{
//    override func viewDidLoad() {
//        super.viewDidLoad()
//        var i=0
//        for vc in self.children{
//            vc.tabBarItem.setTitleTextAttributes([NSAttributedString.Key.foregroundColor : UIColor.redx,NSAttributedString.Key.font : UIFont.small], for: .selected)
//            vc.tabBarItem.image = vc.tabBarItem.image?.withRenderingMode(.alwaysOriginal)
//            vc.tabBarItem.selectedImage = vc.tabBarItem.selectedImage?.withRenderingMode(.alwaysOriginal)
//            vc.tabBarItem.tag = i;
//            i += 1
//        }
//        let tabbar = JuheTabBar()
//        self.setValue(tabbar, forKeyPath: "tabBar")
//        tabbar.centerButton.addTarget(self, action:#selector(centerClicked), for: .touchUpInside)
//    }
//    @objc private func centerClicked(){
//        selectedIndex = 2
//    }
//}

final class MicroEnergyTabBarController: BaseTabBarViewController,UITabBarControllerDelegate {
    let myTabBar = EnergySuperMarketTabBarController.sharedRootViewController
    override func viewDidLoad() {
        super.viewDidLoad()
        self.tabBar.barTintColor = UIColor.thin
        myTabBar.hidesBottomBarWhenPushed = true
        myTabBar.superTabbar = self;
       var i=0
        for vc in self.children{
            vc.tabBarItem.setTitleTextAttributes([NSAttributedString.Key.foregroundColor : UIColor.redx,NSAttributedString.Key.font : UIFont.small], for: .selected)
            vc.tabBarItem.image = vc.tabBarItem.image?.withRenderingMode(.alwaysOriginal)
            vc.tabBarItem.selectedImage = vc.tabBarItem.selectedImage?.withRenderingMode(.alwaysOriginal)
            vc.tabBarItem.tag = i;
            i += 1
        }
    }
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
    }
    override func tabBar(_ tabBar: UITabBar, didSelect item: UITabBarItem) {
        if item.tag != 3{
            myTabBar.backTab = item.tag
        }else if let navi:EnergyNavigationController = children[3] as? EnergyNavigationController{
            navi.pushViewController(myTabBar, animated: true)
        }
    }
}
